org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -579,14 +579,29 @@ private void addResourceDirs(IClasspathDescriptor classpath, IProject project, M
   }
 
   private void addResourceFolder(IClasspathDescriptor classpath, IPath resourceFolder, IPath outputPath,
-      boolean addTestFlag) {
+      boolean addTestFlag, Resource resource) {
     log.info("Adding resource folder " + resourceFolder);
-    IClasspathEntryDescriptor descriptor = classpath.addSourceEntry(resourceFolder, outputPath, DEFAULT_INCLUSIONS,
-        new IPath[] {IPath.fromOSString("**")}, false /*optional*/);
+    IClasspathEntryDescriptor descriptor = classpath.addSourceEntry(resourceFolder, outputPath, 
+      toIPathList(resource.getIncludes(), null),
+      toIPathList(resource.getExcludes(), "**/*.java"), false /*optional*/);
```

Comment:
Can you please explain why you changed the default exclusion pattern? In a quick search I didn't found a reference for that. Please provide a link, in case that's the Maven default exclusion.

Reply:
On my some project, the maven builder often not work, so I had to change the exclusion pattern.

After the change, we don't need maven builder anymore

Reply:
I'm sorry that it does not work for your projects, but M2E is a tool used by a lot of users and has to be as consistent as possible to Maven CLI invocations.
And if standard maven does not define `**/*.java` as default exclusion we won't do it in m2e.

So if you want to exclude it, you have to define it in your pom.xml. You can wrap that in a profile which is activated if the property `m2e.version` is set. M2E injects the property `m2e.version` when projects are build in the IDE, consequently the profile is not activated if build with CLI Maven.

But in general I wonder why you even add `*.java` files in to your resources folder, they are usually placed in the `src` folder.

Reply:
ok `toIPathList(resource.getExcludes(), "**/*.java")` -> `toIPathList(resource.getExcludes(), null)`